### PR TITLE
fix(quantic): issue with date filter and numeric filters when used with dependant facets

### PR DESCRIPTION
**Pull request by mmitiche**
### The issue

Date filters and numeric filters in timeframe facets and numeric facets respectively were not cleared when a parent facet that the numeric and time frame facet depend on is cleared.

The expected behaviour is that when a parent facet is cleared, all the facets that depend on it must also be cleared.

### The source of the issue

The issue was occurring because the numeric and date filters have separate controllers, the `DateFilter` and `NumericFilter` controllers, these controllers are different from the `DateFacet` and `NumericFacet` controllers and these `DateFilter` and `NumericFilter`  controllers create facet objects with a `facet id` different from the one created by `DateFacet` and `NumericFacet` controllers.

Since the facet id is different, the filters were not being take in consideration by  `Facet Condition Manager` and thus they were not cleared when their parent facet is cleared.
This issue has been now fixed, by making the `Facet Condition Manager` aware of both the facets and the filters for numeric and timeframe facets.

### Before:

https://github.com/user-attachments/assets/e015f792-cbeb-404e-a15f-18a6a23aebc3

### After:

https://github.com/user-attachments/assets/110ea0e6-8663-490f-918f-c0285be86fe9

[SFINT-5822]: https://coveord.atlassian.net/browse/SFINT-5822?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/quantic/force-app/main/default/lwc/quanticCategoryFacet/quanticCategoryFacet.js [hunk 1]
```diff
@@ -256,6 +259,7 @@ export default class QuanticCategoryFacet extends LightningElement {
   disconnectedCallback() {
     this.unsubscribe?.();
     this.unsubscribeSearchStatus?.();
+    this.categoryfacetConditionsManager?.stopWatching();
```
Comment (SimonMilord) [inline packages/quantic/force-app/main/default/lwc/quanticCategoryFacet/quanticCategoryFacet.js:262]:
```suggestion
    this.categoryFacetConditionsManager?.stopWatching();
```

File: packages/quantic/force-app/main/default/lwc/quanticCategoryFacet/quanticCategoryFacet.js [hunk 2]
```diff
@@ -327,15 +331,13 @@ export default class QuanticCategoryFacet extends LightningElement {
   }
 
   initFacetConditionManager(engine) {
-    this.facetConditionsManager = this.headless.buildFacetConditionsManager(
-      engine,
-      {
+    this.categoryfacetConditionsManager =
```
Comment (SimonMilord) [inline packages/quantic/force-app/main/default/lwc/quanticCategoryFacet/quanticCategoryFacet.js:334]:
```suggestion
    this.categoryFacetConditionsManager =
```

File: packages/quantic/force-app/main/default/lwc/quanticCategoryFacet/quanticCategoryFacet.js [hunk 3]
```diff
@@ -219,6 +220,8 @@ export default class QuanticCategoryFacet extends LightningElement {
   focusShouldBeInFacet = false;
   /** @type {boolean} */
   hasInitializationError = false;
+  /** @type {FacetConditionsManager} */
+  categoryfacetConditionsManager;
```
Comment (SimonMilord) [inline packages/quantic/force-app/main/default/lwc/quanticCategoryFacet/quanticCategoryFacet.js:224]:
```suggestion
  categoryFacetConditionsManager;
```

**Review by SimonMilord — APPROVED** [review]
Very nice!

**Review by erocheleau — APPROVED** [review]
lgtm

**Pull request merged**